### PR TITLE
add first version of object storage service

### PR DESCRIPTION
**Pull request by wangkirin**
finish the migration of oss and add the backend driver of oss

### CODE REVIEW

**Comment by MabinGo** [comment]
Please delete the redundant git log info:-)

It should be do some changes as below:
1.OSS driver and OSS instance should be separated.
OSS drive should be implemented in Backend,including config and interface for user.
OSS instance should be implemented in OSS package,including boot and storage etc.
2.OSS router should be moved into Dockyard router.go.

File: backend/drivers/oss/oss.go [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+	dataBody, err := ioutil.ReadAll(resp.Body)
+	defer resp.Body.Close()
+	if err != nil {
+		return nil, resp.StatusCode, err
+	}
+	return dataBody, resp.StatusCode, nil
+}
+
+func postFile(path string, data []byte, index int, begin int64, end int64, isLast bool) error {
+	apiserveraddr := fmt.Sprintf("http://0.0.0.0:%v", setting.APIPort)
```
Comment (MabinGo) [inline backend/drivers/oss/oss.go]: apiserveraddr should be generated by config,support http and https

**Comment by wangkirin** [comment]
@MabinGo 
I have updated the PR and the oss can support HTTPS, thank you

**Comment by philips** [comment]
What is OSS? The oss/README.md isn't very helpful.

**Comment by MabinGo** [comment]
@philips OSS offers object storage services in Dockyard,it is in the stage of development now.

**Comment by wangkirin** [comment]
@philips 
Thank you for your attention and your advice! OSS =Object Storage Service, it provides simple distributed backend image storage solution .It  supports  the feature of high availability, high performance and elastic adjustable . Currently README is a note of settings used by oss , we will update it soon to display more details like architecture and how to use.

**Comment by philips** [comment]
So, OSS is itself a distributed replicated object store? Looking forward to
the README :-P

**Comment by wangkirin** [comment]
@MabinGo 
OSS routermoved into Dockyard router.go have finished, thank you!

**Comment by wangkirin** [comment]
@philips 
README have already updated , suggestions are welcome : )

**Comment by wangkirin** [comment]
@MabinGo @philips 
To make the PR looks clean,  I have opened a new PR #82 , you can leave comments there, Thank you : )

**Pull request closed without merging**